### PR TITLE
Bump gradle to 7.3.3, compat java 17

### PR DESCRIPTION
**Pull request by Shoothzj**
### Motivation

Bump gradle to 7.3.1, compat java 17.

### CODE REVIEW

**Review by merlimat — APPROVED** [review]

**Comment by nicoloboschi** [comment]
@Shoothzj any chance you have time to finish this PR ?

**Comment by Shoothzj** [comment]
rerun failure checks

File: dependencies.gradle [hunk 1]
```diff
@@ -25,7 +25,7 @@ depVersions = [
     arquillianCubeDocker: "1.18.2",
     arquillianJunit: "1.6.0.Final",
     bcFips: "1.0.2.1",
-    bouncycastle: "1.69",
```
Comment (nicoloboschi) [inline dependencies.gradle:28]: This should be done in a different pull

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

**Comment by Shoothzj** [comment]
rerun failure checks

**Comment by Shoothzj** [comment]
rerun failure checks

**Comment by Shoothzj** [comment]
rerun failure checks

**Review by nicoloboschi — APPROVED** [review]

**Comment by Shoothzj** [comment]
@merlimat @eolivelli PTAl

**Review by merlimat — APPROVED** [review]

**Pull request merged**